File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.py
def solution(new_id):
    new_id = new_id.lower()
    string = ""
    for i in new_id:
        if i.isalpha() or i.isdigit() or i == "-" or i == "_" or i == ".":
            if i == "." and len(string) > 0 and string[-1] == ".":
                continue
            string += i
    if string[0] == "." and len(string) == 1:
        string = ""
    else:
        if string[0] == ".":
            string = string[1:]
        if string[-1] == ".":
            string = string[:-1]
    if len(string) == 0:
        string = "a"
    if len(string) >= 16:
        string = string[:15]
        if string[0] == ".":
            string = string[1:]
        if string[-1] == ".":
            string = string[:-1]
    while len(string) < 3:
        string += string[-1]
    
    print(string)
    return string

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.java
import java.util.*;

class Solution {
    public String solution(String new_id) {
        new_id = new_id.toLowerCase();
        
        StringBuilder sb = new StringBuilder();
        boolean prevDot = false;
        
        for (int i = 0; i < new_id.length(); i++) {
            char c = new_id.charAt(i);
            if (!((c >= 'a' && c <= 'z') || (c >= '0' && c <= '9') || c == '-' || c == '_' || c == '.')) continue;
            
            if (c == '.') {
                if (prevDot) continue;
                prevDot = true;
            } else {
                prevDot = false;
            }
            
            sb.append(c);
        }
        
        if (sb.length() > 0 && sb.charAt(0) == '.') sb.deleteCharAt(0);
        if (sb.length() > 0 && sb.charAt(sb.length()-1) == '.') sb.deleteCharAt(sb.length()-1);
        
        if (sb.length() == 0) sb.append('a');
        
        if (sb.length() >= 16) {
            sb.setLength(15);
            if (sb.charAt(14) == '.') sb.deleteCharAt(14);
        }
        
        while (sb.length() < 3) {
            sb.append(sb.charAt(sb.length()-1));
        }
        
        return sb.toString();
    }
}